### PR TITLE
🔧 Fix(Fragments/Leptos): Fix unwanted refresh and update to Leptos v0.2

### PR DESCRIPTION
**Pull request by pikhosh**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/create-tauri-app/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### Other information
Nothing...

### CODE REVIEW

File: packages/cli/fragments/fragment-leptos/_Cargo.toml [hunk 1]
```diff
@@ -5,11 +5,13 @@ edition = "2021"
...
-serde-wasm-bindgen = "0.4"
-wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
-wasm-bindgen-futures = "0.4"
+leptos = { version = "0.2.1", features = ["stable"] }
+serde = { version = "1.0.153", features = ["derive"] }
+serde-wasm-bindgen = "0.5.0"
+wasm-bindgen = { version = "0.2.84", features = ["serde-serialize"] }
+wasm-bindgen-futures = "0.4.34"
+js-sys = "0.3.61"
+web-sys = "0.3.61"
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-leptos/_Cargo.toml]:
```suggestion
leptos = { version = "0.2", features = ["stable"] }
serde = { version = "1.0", features = ["derive"] }
serde-wasm-bindgen = "0.5"
wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
wasm-bindgen-futures = "0.4"
js-sys = "0.3"
web-sys = "0.3"
```

File: packages/cli/fragments/fragment-leptos/src/app.rs [hunk 2]
```diff
@@ -60,7 +60,10 @@ pub fn App(cx: Scope) -> impl IntoView {
                 <a href="https://github.com/rust-lang/rust-analyzer" target="_blank">"rust-analyzer"</a>
             </p>
 
-            <form class="row" on:submit=greet>
+            <form class="row" on:submit=move |ev: SubmitEvent| {
+                ev.prevent_default()
+                greet()
+            }>
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-leptos/src/app.rs]: is there a reason behind this change? I think it should be reverted to have consistency with other templates and just call `ev.prevent_default()` above inside the closure

File: .changes/changes.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+"create-tauri-app": patch
+---
+
+Fix(Fragments/Leptos): Fix unwanted refresh when clicking on the button
+Chore(Fragments/Leptos): Update to Leptos v0.2.1
```
Comment (amrbashir) [inline .changes/changes.md]:
```suggestion
Fix unwanted refresh when clicking on the greet button in `leptos` template
```
and add `Update Leptos to v0.2` in another change file

**Review by amrbashir — CHANGES_REQUESTED** [review]
thank you

File: packages/cli/fragments/fragment-leptos/src/app.rs (hunk 2 above)
Comment (pikhosh) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: Ummm... I don't know why I didn't think of that... yeah that makes more sense... :|

File: packages/cli/fragments/fragment-leptos/_Cargo.toml [hunk 4]
```diff
@@ -5,11 +5,13 @@ edition = "2021"
...
 [dependencies]
-leptos = { version = "0.1.0", features = ["stable"] }
+leptos = { version = "0.2", features = ["stable"] }
 serde = { version = "1.0", features = ["derive"] }
-serde-wasm-bindgen = "0.4"
+serde-wasm-bindgen = "0.5"
 wasm-bindgen = { version = "0.2", features = ["serde-serialize"] }
 wasm-bindgen-futures = "0.4"
+js-sys = "0.3"
+web-sys = "0.3"
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-leptos/_Cargo.toml]: I don't think these are needed

File: packages/cli/fragments/fragment-leptos/src/app.rs [hunk 5]
```diff
@@ -1,4 +1,4 @@
-use leptos::web_sys::{Event, SubmitEvent};
+use web_sys::{Event, SubmitEvent};
```
Comment (amrbashir) [inline packages/cli/fragments/fragment-leptos/src/app.rs]:
```suggestion
use leptos::leptos_dom::ev::{InputEvent, SubmitEvent};
```
and update the `update_name` closure to take `InputEvent` instead

**Review by amrbashir — CHANGES_REQUESTED** [review]

File: packages/cli/fragments/fragment-leptos/_Cargo.toml (hunk 4 above)
Comment (pikhosh) [reply packages/cli/fragments/fragment-leptos/_Cargo.toml]: `wasm_bindgen` need `js-sys`.

Comment (amrbashir) [reply packages/cli/fragments/fragment-leptos/_Cargo.toml]: keep `js-sys` then and remove `web-sys` if it is not needed

File: packages/cli/fragments/fragment-leptos/src/app.rs (hunk 5 above)
Comment (pikhosh) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: Apparently `on:input` event type is `Event` and I get an error when using `InputEvent`...

Comment (amrbashir) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: Seems like leptos v0.2 is missing a re-export of `Event` type, Remove the `Event` import all together and change the on:input handler to 
```rs
on:input=move |ev| {
  let v = event_target_value(&ev);
  update_name(v)
}
```
then update `update_name` function to take `String` type instead

Comment (pikhosh) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: Why not just remove the type from the `update_name` closure?

Comment (amrbashir) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: if it works, yes but last time I created the template it wasn't able to infer the type.

Comment (pikhosh) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: Works for me.

Comment (amrbashir) [reply packages/cli/fragments/fragment-leptos/src/app.rs]: cool, lets go with that then

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**